e2e/test/scenarios/sharing/sharing-reproductions.cy.spec.js

```diff
@@ -329,54 +332,57 @@ describe("issue 21559", { tags: "@external" }, () => {
     },
     display: "scalar",
   };
+
   beforeEach(() => {
     restore();
     cy.signInAsAdmin();
 
     setupSMTP();
 
-    cy.createQuestionAndDashboard({
+    createQuestionAndDashboard({
       questionDetails: q1Details,
     }).then(({ body: { dashboard_id } }) => {
-      cy.createQuestion(q2Details);
+      createQuestion(q2Details);
 
       visitDashboard(dashboard_id);
+      cy.findByTestId("scalar-value").should("have.text", "80.52");
       editDashboard();
     });
   });
 
-  it(
-    "should respect dashboard card visualization (metabase#21559)",
-    { tags: "@flaky" },
-    () => {
-      cy.findByTestId("add-series-button").click({ force: true });
+  it("should respect dashboard card visualization (metabase#21559)", () => {
+    cy.findByTestId("add-series-button").click({ force: true });
 
-      // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
+    cy.findByTestId("add-series-modal").within(() => {
       cy.findByText(q2Details.name).click();
-      cy.findByTestId("add-series-modal").within(() => {
-        // wait for elements to appear inside modal
-        chartPathWithFillColor("#A989C5").should("have.length", 1);
-        chartPathWithFillColor("#88BF4D").should("have.length", 1);
 
-        cy.button("Done").click();
-      });
+      // wait for elements to appear inside modal
+      chartPathWithFillColor("#A989C5").should("have.length", 1);
+      chartPathWithFillColor("#88BF4D").should("have.length", 1);
+
+      cy.button("Done").click();
+    });
+
+    cy.findByTestId("add-series-modal").should("not.exist");
```

Comment:
This is an important addition to this test.
Without it, and without scoping the following `chartPathWithFillColor` assertions to the dashboard card, we cannot be sure that Cypress didn't assert that these bars exist in the modal again.

The chances for that were small (because of the length assertion) but not non-existent.